src/main/java/com/bookstore/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.bookstore.service;
+
+import com.bookstore.dto.order.OrderRequestDto;
+import com.bookstore.dto.order.OrderResponseDto;
+import com.bookstore.dto.order.UpdateOrderStatusRequestDto;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.OrderMapper;
+import com.bookstore.model.CartItem;
+import com.bookstore.model.Order;
+import com.bookstore.model.OrderItem;
+import com.bookstore.model.OrderStatus;
+import com.bookstore.model.ShoppingCart;
+import com.bookstore.model.User;
+import com.bookstore.repository.OrderRepository;
+import com.bookstore.repository.ShoppingCartRepository;
+import com.bookstore.repository.UserRepository;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+
+    @Override
+    public OrderResponseDto placeOrder(OrderRequestDto requestDto) {
+        User user = getCurrentUser();
+        ShoppingCart cart = shoppingCartRepository.findByUser(user)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Shopping cart with user id" + user.getId() + "not found"));
+
+        if (cart.getCartItems().isEmpty()) {
+            throw new IllegalStateException("Shopping cart is empty");
```

Comment:
provide parameter to message
```suggestion
            throw new OrderProcessingException("Shopping cart is empty");
```